content/en/docs/tasks/configure-pod-container/user-namespaces.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Use a user namespace with a stateless pod
+reviewers:
+content_type: task
+weight: 160
+---
+
+<!-- overview -->
+{{< feature-state for_k8s_version="v1.25" state="alpha" >}}
+
+This page shows how to configure a user namespace for stateless pods. This
+allows to isolate the user running inside the container from the one in the
+host. A process running as root in a container can run as a different (non-root)
+user in the host.
+
+You can use this feature to reduce the damage a compromised container can do to
+the host or other pods in the same node. There are [several security
+vulnerabilities][KEP-vulns] rated HIGH and CRITICAL that were not explitable
+when user namespaces is active. It is expected user namespace will mitigate some
+future vulnerabilities too.
+
+[KEP-vulns]: https://github.com/kubernetes/enhancements/tree/217d790720c5aef09b8bd4d6ca96284a0affe6c2/keps/sig-node/127-user-namespaces#motivation
+
+## {{% heading "prerequisites" %}}
+
+{{< include "task-tutorial-prereqs.md" >}}
+
+Furthermore, you also need support in your container runtime to use this feature
+with Kubernetes stateless pods:
+
+* CRI-O: v1.25 has support for user namespaces.
+
+* containerd: support is planned for the 1.7 release. See containerd
+  issue [#7063][containerd-userns-issue] for more details.
+
+Support for this in [cri-dockerd is not planned][CRI-dockerd-issue] yet.
+
+Please note that if your container runtime doesn't support user namespaces, the
+new `pod.spec` field will be silently ignored and the pod will be created without
+user namespaces.
+
+[CRI-dockerd-issue]: https://github.com/Mirantis/cri-dockerd/issues/74
+[containerd-userns-issue]: https://github.com/containerd/containerd/issues/7063
+
+<!-- steps -->
+
+## Configure a Pod
+
+A user namespace for a stateless pod is enabled setting the `hostUsers` field of
+`.spec` to false. For example:
+
+{{< codenew file="pods/user-namespaces-stateless.yaml" >}}
+
+1. Create the pod on your cluster:
+
+   ```shell
+   kubectl apply -f https://k8s.io/examples/pods/user-namespaces-stateless.yaml
+   ```
+
+1. Attach to the container and run `cat /proc/self/uid_map`:
+
+   ```shell
+   kubectl attach -it userns bash
+   ```
+
+And run the command. The output is similar to this:
+
+```none
+root@userns:/# cat /proc/self/uid_map
+         0     131072      65536
+```
+
+This means that UID 0 is mapped to UID 131072 in the host and this is done for
+65536 consecutive UIDs. In other words, UID 1 in the container is mapped to UID
+131072+1 on the host, and so on.
+
+For more details, please see `man 7 user_namespaces`.
+
+## Verify if user namespaces is active for a pod
+
+To verify that a pod is using user namespaces, if you exec into the pod you can
+run:
+
+```none
+root@userns:/# cat /proc/self/uid_map
+         0     131072      65536
+root@userns:/# cat /proc/self/gid_map
+         0     131072      65536
```

Comment:
Ideally, [omit the prompt](https://kubernetes.io/docs/contribute/style/style-guide/#don-t-include-the-command-prompt).

Reply:
Fixed